src/Renci.SshNet/Security/KeyExchangeEC.BclImpl.cs

```diff
@@ -4,9 +4,9 @@
 
 namespace Renci.SshNet.Security
 {
-    internal abstract partial class KeyExchangeECDH
+    internal abstract partial class KeyExchangeEC
     {
-        private sealed class BclImpl : Impl
+        protected internal sealed class BclImpl : Impl
```

Comment:
nit, here and elsewhere

```suggestion
        protected sealed class BclImpl : Impl
```